compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs

```diff
@@ -965,35 +966,39 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         let expr = &self.infcx.tcx.hir().expect_expr(hir_id).kind;
         debug!("closure_span: hir_id={:?} expr={:?}", hir_id, expr);
         if let hir::ExprKind::Closure(.., body_id, args_span, _) = expr {
-            for (upvar_hir_id, place) in
-                self.infcx.tcx.typeck(def_id.expect_local()).closure_captures[&def_id]
-                    .keys()
-                    .zip(places)
+            for (captured_place, place) in self
+                .infcx
+                .tcx
+                .typeck(def_id.expect_local())
+                .closure_min_captures_flattened(def_id)
+                .zip(places)
             {
-                let span = self.infcx.tcx.upvars_mentioned(local_did)?[upvar_hir_id].span;
+                let upvar_hir_id = match captured_place.place.base {
+                    //FIXME(project-rfc-2229#8): Use better span from captured_place
+                    HirPlaceBase::Upvar(upvar_id) => upvar_id.var_path.hir_id,
+                    base => bug!("Expected upvar, found={:?}", base),
+                };
+                let span =
```

Comment:
why did u change this?

Reply:
![image](https://user-images.githubusercontent.com/10948303/103467067-0c406500-4d19-11eb-8896-ebd4fdebd8fd.png)
ran into this issue and this change fixed it

Reply:
Ah okay, you just need `&upvar_hir_id` there btw.